paasta_tools/cli/utils.py

```diff
@@ -507,7 +508,9 @@ def run_paasta_serviceinit(subcommand, master, service, instances, cluster, stre
     command_without_empty_strings = [part for part in command_parts if part != '']
     command = ' '.join(command_without_empty_strings)
     log.debug("Running Command: %s" % command)
-    _, output = _run(command, timeout=timeout, stream=stream)
+    return_code, output = _run(command, timeout=timeout, stream=stream)
+    if return_code != 0:
+        raise CalledProcessError(return_code, command, output)
```

Comment:
>I am not sure we need to raise an exception on the client side (with a not quite related stack trace) or to simply exit with the error code.

I think a good compromise is to just catch this exception in the `paasta status` handler and just print out a small message at the end informing the user that `ssh ...` returned with a non zero return code.